service/service.go

```diff
@@ -137,14 +144,22 @@ func (s *Service) prepareBeforeStart() {
 	http.Handle("/", s.Router)
 }
 
-// HandleFunc wrapper with token validation, logging recovery and metrics
+// HandleFunc excepts a HanderFunc an converts it to a handler, then registers this handler
 func (s *Service) HandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) *mux.Route {
 	return s.Handle(pattern, http.HandlerFunc(handler))
 }
 
-// Handle
+// Handle is a wrapper around the original Go handle func with logging recovery and metrics
 func (s *Service) Handle(pattern string, originalHandler http.Handler) *mux.Route {
 	h := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
```

Comment:
OK. So the `recover` is the trick. I assume this is always called, even if there's no pani, but then the `err` is nil and the rest won't get executed. Is that about right?

Reply:
Yes. see https://blog.golang.org/defer-panic-and-recover